hydra-cluster/test/Test/EndToEndSpec.hs

```diff
@@ -487,6 +497,76 @@ waitForLog delay nodeOutput failureMessage predicate = do
         ]
           <> logs
 
+timedTx :: FilePath -> Tracer IO EndToEndLog -> RunningNode -> TxId -> IO ()
+timedTx tmpDir tracer node@RunningNode{nodeSocket} hydraScriptsTxId = do
+  (aliceCardanoVk, aliceCardanoSk) <- keysFor Alice
+  let aliceSk = generateSigningKey "alice-timed"
+  let alice = deriveParty aliceSk
+  let contestationPeriod = UnsafeContestationPeriod 2
+  aliceChainConfig <- chainConfigFor Alice tmpDir nodeSocket [] contestationPeriod
+  withHydraNode tracer aliceChainConfig tmpDir 1 aliceSk [] [1] hydraScriptsTxId $ \n1 -> do
+    waitForNodesConnected tracer [n1]
+    now <- getCurrentTime
+
+    -- Funds to be used as fuel by Hydra protocol transactions
+    seedFromFaucet_ node aliceCardanoVk 100_000_000 Fuel (contramap FromFaucet tracer)
+    send n1 $ input "Init" []
+    headId <-
+      waitForAllMatch 10 [n1] $
+        headIsInitializingWith (Set.fromList [alice])
+
+    -- Get some UTXOs to commit to a head
+    committedUTxOByAlice <- seedFromFaucet node aliceCardanoVk aliceCommittedToHead Normal (contramap FromFaucet tracer)
+    send n1 $ input "Commit" ["utxo" .= committedUTxOByAlice]
+    waitFor tracer 3 [n1] $ output "HeadIsOpen" ["utxo" .= committedUTxOByAlice, "headId" .= headId]
+
+    -- Acquire a current point in time
+    now' <- getCurrentTime
+    let slotLengthMillis = 100
+
+    -- Create an arbitrary transaction using some input.
+    let paymentFromAliceToFaucet :: Num a => a
+        paymentFromAliceToFaucet = 1_000_000
+    let firstCommittedUTxO = Prelude.head $ UTxO.pairs committedUTxOByAlice
+    (faucetVk, _) <- keysFor Faucet
```

Comment:
Could: simplify the transaction further.

Have alice pay funds to herself would be even more simple and not require this statement for example.